### PR TITLE
Add language switch to request examples

### PR DESCRIPTION
**Pull request by albert-carreras**
### Adds a convenient language switcher to the request examples

- Adds new code block directive `code_block_selector_directive.py` that uses `.. code-block-selector::`  to detect the need for the selector and adds a few html elements:
   - A switcher container
   - An empty div (with `nodes.raw`) which only purpose is to have the `data-enhancer` attribute visible
   - The Switcher buttons
   - The code-blocks with new ids to find them on the DOM

- Adds a new data-enhancer handler that deals with the hiding and showing of codeblocks, remembering selected language among others.
- Adds the new directive to all relevant `.rst` files
- Improves responsiveness in all pages

### CODE REVIEW

File: source/reference/v2/captures-api/get-capture.rst [hunk 1]
```diff
@@ -152,25 +152,22 @@ Response
...
-^^^^^^^^^^^^^
-.. code-block:: php
-   :linenos:
+   .. code-block:: php
+    :linenos:
 
-    <?php
-    $mollie = new \Mollie\Api\MollieApiClient();
-    $mollie->setApiKey("test_dHar4XY7LxsDOtmnkVtjNVWXLSlXsM");
+        <?php
```
Comment (mvdpanne) [inline source/reference/v2/captures-api/get-capture.rst]: This indentation is not necessary/adds space to the code example.

File: source/reference/v2/captures-api/list-captures.rst [hunk 2]
```diff
@@ -103,25 +103,23 @@ Response
...
+    curl -X GET https://api.mollie.com/v2/payments/tr_WDqYK6vllg/captures \
+        -H "Authorization: Bearer test_dHar4XY7LxsDOtmnkVtjNVWXLSlXsM"
+
+   .. code-block:: php
+    :linenos:
 
-    <?php
-    $mollie = new \Mollie\Api\MollieApiClient();
-    $mollie->setApiKey("test_dHar4XY7LxsDOtmnkVtjNVWXLSlXsM");
+        <?php
```
Comment (mvdpanne) [inline source/reference/v2/captures-api/list-captures.rst]: Please check all PHP examples for this additional indentation.

File: source/reference/v2/captures-api/get-capture.rst [hunk 3]
```diff
@@ -152,25 +152,22 @@ Response
 Example
 -------
 
-Request (curl)
-^^^^^^^^^^^^^^
-.. code-block:: bash
-   :linenos:
+.. code-block-selector::
+   .. code-block:: bash
+    :linenos:
```
Comment (mvdpanne) [inline source/reference/v2/captures-api/get-capture.rst]: The `:linenos:` does not seem to be aligned correctly. The `Response` examples on show the correct indentation of the directive arguments and body. You can also see the example here: https://mollie.atlassian.net/browse/PROC-1601

It's kinda a pain to edit it all, but RST is quite sensitive to whitespace, so I'd say it's worth it to keep it strict and clean.

File: source/theme/js/exampleSwitcher.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import { enhance } from './utils';
+
+// Gets all the codeblocks present and hides all of them except the one from the argument
+const hideCodeBlocks = language => {
+  const codeBlocks = Array.from(document.querySelectorAll('[id^="request-"]'));
+  codeBlocks.forEach(block => {
+    const elementId = `request-${language.toLowerCase()}`;
+    if (block.id !== elementId) block.classList.add('hidden');
```
Comment (mvdpanne) [inline source/theme/js/exampleSwitcher.js]: Is it default code style in JS not to use curly braces for conditional statements?

File: source/theme/js/exampleSwitcher.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    else block.classList.remove('hidden');
+  });
+};
+
+export default enhance('example-switcher', element => {
+  // Gets stored language if present
+  const storedLanguage = localStorage.getItem('preferedLanguage');
+  const language = storedLanguage && storedLanguage !== 'undefined' ? storedLanguage : 'curl';
```
Comment (mvdpanne) [inline source/theme/js/exampleSwitcher.js]: This feels a little crammed together, but maybe that's just me.

File: source/theme/js/exampleSwitcher.js (hunk 4 above)
Comment (vernondegoede) [reply source/theme/js/exampleSwitcher.js]: Nope, we always use curly brackets at Mollie. Can you make sure this uses curly brackets as well?

File: source/theme/js/exampleSwitcher.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+      block.classList.add('hidden');
+    } else {
+      block.classList.remove('hidden');
+    }
+  });
+};
+
+export default enhance('example-switcher', element => {
+  // Gets stored language if present
+  const storedLanguage = localStorage.getItem('preferedLanguage');
```
Comment (vernondegoede) [inline source/theme/js/exampleSwitcher.js]: It should be `preferredLanguage` 🥇

File: source/theme/js/exampleSwitcher.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  let language;
+  if (storedLanguage && storedLanguage !== 'undefined') {
+    language = storedLanguage;
+  } else {
+    language = 'curl';
+  }
+  const selectedLanguage = document.getElementById(`example-switch-${language}`);
```
Comment (vernondegoede) [inline source/theme/js/exampleSwitcher.js]: Can you add a check here and see if the element is available? If not, fall back to `DEFAULT_LANGUAGE` (which you can re-use in the `} else {` on line 23.

Comment (vernondegoede) [reply source/theme/js/exampleSwitcher.js]: The reason we need to check this, is because I might have selected "Ruby" and navigate to a page where there is no Ruby example.

File: source/theme/styles/components/_example.scss [hunk 8]
```diff
@@ -1,3 +1,49 @@
...
+      &.selected {
+        font-weight: 600;
+        border-bottom: none;
+      }
+    }
+  }
+
+  .code-blocks > div {
+
+    &:not(.hidden) {
```
Comment (vernondegoede) [inline source/theme/styles/components/_example.scss]: Can you use `.is-hidden` here? Hidden is a stateful classname, so it's allowed to break the BEM rules.

File: source/theme/styles/components/_example.scss [hunk 9]
```diff
@@ -1,3 +1,49 @@
...
+      cursor: pointer;
+
+      &:hover,
+      &:focus {
+        color: $brand-primary-hover;
+        text-decoration: none;
+        border-bottom: none;
+      }
+
+      &.selected {
```
Comment (vernondegoede) [inline source/theme/styles/components/_example.scss]: `.is-selected`

File: source/theme/styles/components/_example.scss [hunk 10]
```diff
@@ -1,3 +1,49 @@
+.rst-content #example {
+  max-width: 700px;
+
+  .examples-switcher {
+
```
Comment (vernondegoede) [inline source/theme/styles/components/_example.scss]: Nitpick: Can you remove the whitespace? :)

File: source/theme/styles/components/_example.scss [hunk 11]
```diff
@@ -1,3 +1,49 @@
...
+      &.selected {
+        font-weight: 600;
+        border-bottom: none;
+      }
+    }
+  }
+
+  .code-blocks > div {
+
```
Comment (vernondegoede) [inline source/theme/styles/components/_example.scss]: Nitpick: Unnecessary whitespace

File: source/theme/js/exampleSwitcher.js [hunk 12]
```diff
@@ -0,0 +1,46 @@
+import { enhance } from './utils';
+
+// Gets all the codeblocks present and hides all of them except the one from the argument
+const hideCodeBlocks = language => {
+  const codeBlocks = Array.from(document.querySelectorAll('[id^="request-"]'));
+  codeBlocks.forEach(block => {
+    const elementId = `request-${language.toLowerCase()}`;
+    if (block.id !== elementId) {
+      block.classList.add('hidden');
```
Comment (vernondegoede) [inline source/theme/js/exampleSwitcher.js]: Can you also add an `aria-hidden` attribute here?

File: source/theme/styles/main.scss [hunk 13]
```diff
@@ -188,10 +188,10 @@ a.headerlink::before {
 @import 'components/code';
 @import 'components/content';
 @import 'components/endpoint';
+@import 'components/example';
```
Comment (vernondegoede) [inline source/theme/styles/main.scss]: Could you rename this to `code-example`?

**Comment by vernondegoede** [comment]
`localStorage` is not available in Safari private mode if I remember correctly. Can we add a check for this and always just show the first item (instead of `getItem`) if it isn't available?

More information: https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API/Using_the_Web_Storage_API#Testing_for_availability

File: source/orders/overview.rst [hunk 14]
```diff
@@ -14,7 +14,7 @@ How does the Orders API work?
 #. For every order in your webshop, you create an order using the
    :doc:`Create Order API </reference/v2/orders-api/create-order>`.
 
-#. The :ref:`Create Order API response <get-order-response>` contains the ``_links.checkout`` property. This is a link
+#. The :ref:`Create Order API response <create-order-response>` contains the ``_links.checkout`` property. This is a link
```
Comment (vernondegoede) [inline source/orders/overview.rst]: Sharp 💯

File: source/theme/js/exampleSwitcher.js (hunk 6 above)
Comment (albert-carreras) [reply source/theme/js/exampleSwitcher.js]: 👀

**Comment by mvdpanne** [comment]
Ah, one last thing I just noticed: the width of the code examples does always not match up with the width of the parameter table. When you widen your browser, the code example does not get as wide as the parameter table.
